sshcode.go

```diff
@@ -53,6 +55,41 @@ func sshCode(host, dir string, o options) error {
 		return xerrors.Errorf("failed to find available remote port: %w", err)
 	}
 
+	// Start SSH master connection socket. This prevents multiple password prompts from appearing as authentication
+	// only happens on the initial connection.
+	var sshMasterCmd *exec.Cmd
+	if !o.noReuseConnection {
+		newSSHFlags := fmt.Sprintf(`%v -o "ControlPath=%v"`, o.sshFlags, sshControlPath)
+
+		// -MN means "start a master socket and don't open a session, just connect".
+		sshMasterCmdStr := fmt.Sprintf(`ssh %v -MN %v`, newSSHFlags, host)
+		sshMasterCmd = exec.Command("sh", "-c", sshMasterCmdStr)
```

Comment:
if you create a cancellable context and pass it to `exec.CommandContext` you don't have to manually do the process cleanup

Reply:
If the master process gets killed it leaves the socket file behind which prevents the master from creating the socket next time.

Reply:
I've removed the tidyup code at the bottom of the `sshCode` function into a simpler defer statement. Let me know if this isn't what you want.